src/theory/uf/diamonds_proof_generator.cpp

```diff
@@ -167,10 +169,77 @@ void DiamondsProofGenerator::ppStaticLearn(TNode n,
 
 std::shared_ptr<ProofNode> DiamondsProofGenerator::getProofFor(Node fact)
 {
-  Trace("valid-witness") << "Prove " << fact << std::endl;
-  // proofs not yet supported
+  NodeManager* nm = nodeManager();
   CDProof cdp(d_env);
-  cdp.addTrustedStep(fact, TrustId::DIAMONDS, {}, {});
+  Trace("diamonds-proof") << "Prove " << fact << std::endl;
+  Assert(fact.getKind() == Kind::IMPLIES);
+  Node antec = fact[0];
+  Assert(antec.getKind() == Kind::OR);
+  Node conc = fact[1];
+  Assert(conc.getKind() == Kind::EQUAL);
+  std::vector<Node> children(fact.begin(), fact.end());
```

Comment:
It's a little confusing that `children` is initialized here with `{antec,conc}` just to be replaced by `{antec[0],conc}` before being used. I think the comment I describe above might resolve this.